### DISCUSSION TITLE
CoAP protocol

### DISCUSSION THREAD

**Discussion by edgargom** [ankraft/ACME-oneM2M-CSE#151] [General]
Hello Ankraft,

I'm trying to explore the ACME implementation of OneM2M and trying to use it with CoAP protocol and I can see that in configuration files it seems to support CoAP but in documentation I can't find anything about it... Does ACME support CoAP?

Thanks in advance.

**Comment by ankraft** [comment]
Thank you for your question. 

Unfortunately, CoAP is yet not supported, but in development. I hope to finish a first version in the next couple of months. 

I started an implementation on a CoAP binding a while ago but stopped because we (that is the oneM2M SDS working group) were discussing a couple of breaking changes in the CoAP binding specification and I wanted to wait until the dust settled. I left the CoAP settings in the configuration because I will continue from there.

That said, there is binding for CoAP in the branch "CoAP-support" that was contributed about three years ago. 
Be aware that this is based in a rather old version of ACME, but I am sure that you can use it for first experiments.

One question from my side to prioritise the implementation: would you consider DTLS support as an important feature to have for CoAP?

**Reply by edgargom** [reply]
Thank you for your quick feedback.

It's unfortunate to hear that ACME does not yet support CoAP.

I'm sorry I don't think I'm qualified to answer your questions as I'm a student still learning oneM2M and CoAP and the use case that I'm trying to implement doesn't really take into account security as a main priority right now so I haven't though much about that regard...

**Reply by ankraft** [reply]
Perhaps you still might have a look at the old branch. It is a complete implementation that was used within a PoC, though the ACME CSE is an old version, but still supports the basic resource types and is fully compliant. It might be sufficient for your studies.

Good luck with your studies. I'll let you know when the implementation is in a usable state.

**Comment by edgargom** [comment]
Ok, I'll have a look. Thanks!

**Comment by ankraft** [comment]
Update: The CoAP binding is now available: https://github.com/ankraft/ACME-oneM2M-CSE/releases/tag/2024.10
